Update Kubernetes documentation (#98)
hazelcast/hz-docs: docs/modules/deploy/pages/configuring-kubernetes.adoc

== Discovering Members from Hazelcast Clients

For the client to discover the Hazelcast cluster, all it needs to know is the address by which the cluster is accessible.

=== Inside Kubernetes Cluster

> If you have a Hazelcast cluster deployed on Kubernetes and you want to configure Hazelcast Client deployed on the same Kubernetes cluster, then you'll usually use the Kubernetes service name in the client's configuration.

Reviewer: ```suggestion
If you have a Hazelcast cluster and a Hazelcast client deployed on the same Kubernetes cluster, you should use the Kubernetes service name in the client's configuration.
```

```yaml
hazelcast-client:
  network:
    cluster-members:
      - MY-SERVICE-NAME
```
--
Java:: 
+ 
-- 
```java
clientConfig.getNetworkConfig().addAddress("MY-SERVICE-NAME");
```
--
NodeJS:: 
+ 
-- 
```javascript
const clientConfig = {
    network: {
        clusterMembers: [
            'MY-SERVICE-NAME'
        ]
    }
};
```
--
Python:: 
+ 
-- 
```python
client = hazelcast.HazelcastClient(
    cluster_members=["MY-SERVICE-NAME"],
)
```
--
C++:: 
+ 
-- 
```cpp
config.get_network_config().add_address({"MY-SERVICE-NAME", 5701})
```
--
Go:: 
+ 
-- 
```go
config.Cluster.Network.SetAddresses("MY-SERVICE-NAME:5701")
```
--
====

For the complete example, please check link:https://guides.hazelcast.org/kubernetes/[Hazelcast Guides: Hazelcast for Kubernetes].

=== Outside Kubernetes Cluster

> If your Hazelcast cluster is deployed on Kubernetes, but Hazelcast Client is in a completely different network, then it can connect only through the public Internet network. This requires exposing each Hazelast member pod with a dedicated NodePort or LoadBalancer Kubernetes service. For details and a complete example, please check link:https://guides.hazelcast.org/kubernetes-external-client/[Hazelcast Guides: Connect External Hazelcast Client to Kubernetes].

Reviewer: ```suggestion
If your Hazelcast cluster is deployed on Kubernetes, but your Hazelcast client is in a completely different network, then it can connect only through the public Internet. This requires exposing each Hazelast member pod with a dedicated NodePort or LoadBalancer Kubernetes service. For details and a complete example, please check link:https://guides.hazelcast.org/kubernetes-external-client/[Hazelcast Guides: Connect External Hazelcast Client to Kubernetes].
```
